virtcontainers/qemu_arch_base.go

```diff
@@ -458,6 +458,23 @@ func (q *qemuArchBase) appendNetwork(devices []govmmQemu.Device, endpoint Endpoi
 			},
 		)
 		q.networkIndex++
+	case *BridgedMacvlanEndpoint:
+		devices = append(devices,
+			govmmQemu.NetDevice{
+				Type:          networkModelToQemuType(ep.NetPair.NetInterworkingModel),
+				Driver:        govmmQemu.VirtioNetPCI,
+				ID:            fmt.Sprintf("network-%d", q.networkIndex),
+				IFName:        ep.NetPair.TAPIface.Name,
+				MACAddress:    ep.NetPair.TAPIface.HardAddr,
+				DownScript:    "no",
+				Script:        "no",
+				VHost:         true,
```

Comment:
Does `vhost` have to be true for this case?

Also, except vhost, those two structures look identical to me, IMO you could simply do:
```go
switch ep := endpoint.(type) {
 	case *VirtualEndpoint:
                fallthrough
        case *BridgedMacvlanEndpoint:
                ...
```

Reply:
Sorry, I think `fallthrough` is not appropriate here, but I think you can do:
```go
switch ep := endpoint.(type) {
 	case *VirtualEndpoint, *BridgedMacvlanEndpoint:
                ...
```

Reply:
Has been addressed.